internal/layers/layers.go

```diff
@@ -95,22 +101,21 @@ func MountLCOWLayers(ctx context.Context, containerID string, layerFolders []str
 
 	for _, layerPath := range layerFolders[:len(layerFolders)-1] {
 		log.G(ctx).WithField("layerPath", layerPath).Debug("mounting layer")
-		var (
+		if path.Ext(layerPath) != ".vhd" && path.Ext(layerPath) != ".vhdx" {
 			layerPath = filepath.Join(layerPath, "layer.vhd")
-			uvmPath   string
-		)
-		uvmPath, err = addLCOWLayer(ctx, vm, layerPath)
+		}
+		layerMount, err := addLCOWLayer(ctx, vm, layerPath)
```

Comment:
Given `layerMount` is only composed of `layerPath`, which we pass in, and `uvmPath`, which `addLCOWLayer` used to return. It might be simpler to just keep the return the same and construct a `LayerMount` here.

Reply:
Seems reasonable to me :)